From b0ab6c7a59bc28ff6997a16ace2a32ab65a4ac8f Mon Sep 17 00:00:00 2001
From: Thomas Preud'homme <robotux@celest.fr>
Date: Thu, 3 Jan 2013 15:16:14 +0100
Subject: Set resources installation path.

Set the place to install resources (executable, lang files, themes,
plugins).

Origin: vendor
Forwarded: no
Last-Update: 2013-01-03
---
 extratool.pri                                         |    4 ++++
 plugins-alternative/Themes/Clean/interface.pro        |    8 ++++++++
 plugins-alternative/Themes/Teracopy/interface.pro     |    8 ++++++++
 plugins/CopyEngine/Ultracopier-0.3/copyEngine.pro     |    8 ++++++++
 plugins/Listener/catchcopy-v0002/listener.pro         |    5 +++++
 plugins/PluginLoader/catchcopy-v0002/pluginLoader.pro |    5 +++++
 plugins/SessionLoader/KDE4/sessionLoader.pro          |    5 +++++
 plugins/SessionLoader/Windows/sessionLoader.pro       |    5 +++++
 plugins/Themes/Oxygen/interface.pro                   |    8 ++++++++
 ultracopier.pro                                       |    4 ++++
 10 files changed, 60 insertions(+)
 create mode 100644 extratool.pri

diff --git a/extratool.pri b/extratool.pri
new file mode 100644
index 0000000..618456a
--- /dev/null
+++ b/extratool.pri
@@ -0,0 +1,4 @@
+defineReplace(superBaseName) {
+    THEMEDIR = $$dirname($$1)
+    return($$basename(THEMEDIR)/$$basename($$1))
+}
diff --git a/plugins-alternative/Themes/Clean/interface.pro b/plugins-alternative/Themes/Clean/interface.pro
index 4330f74..ca8510d 100644
--- a/plugins-alternative/Themes/Clean/interface.pro
+++ b/plugins-alternative/Themes/Clean/interface.pro
@@ -8,6 +8,8 @@ HEADERS         = interface.h \
 SOURCES         = interface.cpp \
     factory.cpp
 TARGET          = $$qtLibraryTarget(interface)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
 include(../../../updateqm.pri)
 TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/de/translation.ts \
@@ -26,6 +28,12 @@ TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/tr/translation.ts \
     Languages/zh/translation.ts
 
+translations.files = Languages
+translations.path = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target translations infos
+
 FORMS += \
     interface.ui
 
diff --git a/plugins-alternative/Themes/Teracopy/interface.pro b/plugins-alternative/Themes/Teracopy/interface.pro
index 7c05519..f0b5e6f 100644
--- a/plugins-alternative/Themes/Teracopy/interface.pro
+++ b/plugins-alternative/Themes/Teracopy/interface.pro
@@ -11,6 +11,8 @@ HEADERS         = interface.h \
 SOURCES         = interface.cpp \
     factory.cpp
 TARGET          = $$qtLibraryTarget(interface)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
 include(../../../updateqm.pri)
 TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/de/translation.ts \
@@ -29,6 +31,12 @@ TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/tr/translation.ts \
     Languages/zh/translation.ts
 
+translations.files = Languages
+translations.path = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target translations infos
+
 RESOURCES	+= resources.qrc
 
 FORMS += \
diff --git a/plugins/CopyEngine/Ultracopier-0.3/copyEngine.pro b/plugins/CopyEngine/Ultracopier-0.3/copyEngine.pro
index 03a5f3f..20cc890 100644
--- a/plugins/CopyEngine/Ultracopier-0.3/copyEngine.pro
+++ b/plugins/CopyEngine/Ultracopier-0.3/copyEngine.pro
@@ -39,6 +39,8 @@ SOURCES         = copyEngine.cpp \
     copyEngine-collision-and-error.cpp \
     ListThread.cpp
 TARGET          = $$qtLibraryTarget(copyEngine)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
 include(../../../updateqm.pri)
 TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/de/translation.ts \
@@ -58,6 +60,12 @@ TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/tr/translation.ts \
     Languages/zh/translation.ts
 
+translations.files = Languages
+translations.path = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target translations infos
+
 FORMS += \
     options.ui \
     fileErrorDialog.ui \
diff --git a/plugins/Listener/catchcopy-v0002/listener.pro b/plugins/Listener/catchcopy-v0002/listener.pro
index 59a2379..4f141b6 100644
--- a/plugins/Listener/catchcopy-v0002/listener.pro
+++ b/plugins/Listener/catchcopy-v0002/listener.pro
@@ -14,3 +14,8 @@ SOURCES         = listener.cpp \
     catchcopy-api-0002/ServerCatchcopy.cpp \
     catchcopy-api-0002/ExtraSocketCatchcopy.cpp
 TARGET          = $$qtLibraryTarget(listener)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target infos
diff --git a/plugins/PluginLoader/catchcopy-v0002/pluginLoader.pro b/plugins/PluginLoader/catchcopy-v0002/pluginLoader.pro
index 03af533..ad53c76 100755
--- a/plugins/PluginLoader/catchcopy-v0002/pluginLoader.pro
+++ b/plugins/PluginLoader/catchcopy-v0002/pluginLoader.pro
@@ -11,6 +11,11 @@ HEADERS         = \
 SOURCES         = \
     pluginLoader.cpp
 TARGET          = $$qtLibraryTarget(pluginLoader)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target infos
 
 LIBS += -lole32
 
diff --git a/plugins/SessionLoader/KDE4/sessionLoader.pro b/plugins/SessionLoader/KDE4/sessionLoader.pro
index e036663..0f62558 100644
--- a/plugins/SessionLoader/KDE4/sessionLoader.pro
+++ b/plugins/SessionLoader/KDE4/sessionLoader.pro
@@ -8,3 +8,8 @@ HEADERS         = sessionLoader.h \
     ../../../interface/PluginInterface_SessionLoader.h
 SOURCES         = sessionLoader.cpp
 TARGET          = $$qtLibraryTarget(sessionLoader)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target infos
diff --git a/plugins/SessionLoader/Windows/sessionLoader.pro b/plugins/SessionLoader/Windows/sessionLoader.pro
index e036663..0f62558 100644
--- a/plugins/SessionLoader/Windows/sessionLoader.pro
+++ b/plugins/SessionLoader/Windows/sessionLoader.pro
@@ -8,3 +8,8 @@ HEADERS         = sessionLoader.h \
     ../../../interface/PluginInterface_SessionLoader.h
 SOURCES         = sessionLoader.cpp
 TARGET          = $$qtLibraryTarget(sessionLoader)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target infos
diff --git a/plugins/Themes/Oxygen/interface.pro b/plugins/Themes/Oxygen/interface.pro
index 97b44b9..d583420 100644
--- a/plugins/Themes/Oxygen/interface.pro
+++ b/plugins/Themes/Oxygen/interface.pro
@@ -14,6 +14,8 @@ SOURCES         = interface.cpp \
     factory.cpp \
     TransferModel.cpp
 TARGET          = $$qtLibraryTarget(interface)
+include(../../../extratool.pri)
+target.path     = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
 include(../../../updateqm.pri)
 TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/de/translation.ts \
@@ -33,6 +35,12 @@ TRANSLATIONS   += Languages/ar/translation.ts \
     Languages/tr/translation.ts \
     Languages/zh/translation.ts
 
+translations.files = Languages
+translations.path = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+infos.files      = informations.xml
+infos.path       = $${PREFIX}/lib/ultracopier/$$superBaseName(_PRO_FILE_PWD_)
+INSTALLS       += target translations infos
+
 win32 {
     RESOURCES += resources_windows.qrc
 }
diff --git a/ultracopier.pro b/ultracopier.pro
index e69ab7f..6a613bb 100644
--- a/ultracopier.pro
+++ b/ultracopier.pro
@@ -18,9 +18,13 @@ TRANSLATIONS += plugins/Languages/ar/translation.ts \
     plugins/Languages/th/translation.ts \
     plugins/Languages/tr/translation.ts \
     plugins/Languages/zh/translation.ts
+translations.files += plugins/Languages
+translations.path = $${PREFIX}/share/ultracopier
 
 win32:RC_FILE += resources/resources-windows.rc
 TARGET = ultracopier
+target.path = $${PREFIX}/bin
+INSTALLS += target translations
 macx { 
     ICON = resources/ultracopier.icns
 }
